src/EnergyPlus/WaterCoils.cc

```diff
@@ -2042,44 +2007,72 @@ namespace WaterCoils {
 					TempSize = AutoSize;
 					RequestSizing( CompType, CompName, HeatingCoilDesAirOutletTempSizing, SizingString, TempSize, bPRINT, RoutineName );
 					DataDesOutletAirTemp = TempSize;
-					if (CurOASysNum > 0) {
+					if ( CurOASysNum > 0 ) {
 						OASysEqSizing( CurOASysNum ).AirFlow = true;
 						OASysEqSizing( CurOASysNum ).AirVolFlow = FinalSysSizing( CurSysNum ).DesOutAirVolFlow;
 					}
 					TempSize = AutoSize; // reset back
 				}
 				RequestSizing( CompType, CompName, HeatingAirflowSizing, SizingString, TempSize, bPRINT, RoutineName );
 				// reset the design air volume flow rate for air loop coils only
-				if (CurSysNum > 0) WaterCoil( CoilNum ).DesAirVolFlowRate = TempSize;
+				if ( CurSysNum > 0 ) WaterCoil( CoilNum ).DesAirVolFlowRate = TempSize;
 				WaterCoil( CoilNum ).InletAirMassFlowRate = StdRhoAir * TempSize; // inlet air mass flow rate is not the autosized value
 				DataAirFlowUsedForSizing = TempSize;
 				DataFlowUsedForSizing = TempSize; // many autosized inputs use the design (autosized) air flow rate, save this value
 
 				bPRINT = true;
-				TempSize = AutoSize;
-				SizingString = "Design Coil Load [W]";
+				if ( WaterCoil( CoilNum ).CoilPerfInpMeth == NomCap && NomCapUserInp ) {
+					TempSize = WaterCoil( CoilNum ).DesTotWaterCoilLoad;
+					DataNomCapInpMeth = true;
+				} else {
+					TempSize = AutoSize;
+				}
 				if ( CurSysNum > 0 ) {
 					SizingType = HeatingCapacitySizing;
 				} else {
 					SizingType = WaterHeatingCapacitySizing;
 				}
+				FieldNum = 3; //  N3 , \field Rated Capacity
+				SizingString = WaterCoilNumericFields( CoilNum ).FieldNames( FieldNum ) + " [W]";
 				RequestSizing( CompType, CompName, SizingType, SizingString, TempSize, bPRINT, RoutineName );
 				WaterCoil( CoilNum ).DesWaterHeatingCoilRate = TempSize;
-				if (( WaterCoil( CoilNum ).CoilPerfInpMeth == UAandFlow && WaterCoil( CoilNum ).UACoil == AutoSize ) || ( WaterCoil( CoilNum ).CoilPerfInpMeth == NomCap && WaterCoil( CoilNum ).DesTotWaterCoilLoad == AutoSize )) {
-					// WaterCoil ( CoilNum ).DesTotWaterCoilLoad will be set below in UA calc
-				} else {
-					if (WaterCoil( CoilNum ).DesTotWaterCoilLoad == AutoSize) WaterCoil( CoilNum ).DesTotWaterCoilLoad = TempSize;
+				WaterCoil( CoilNum ).DesTotWaterCoilLoad = TempSize;
+				DataCapacityUsedForSizing = WaterCoil( CoilNum ).DesWaterHeatingCoilRate;
+
+				// We now have the design load if it was autosized. For the case of CoilPerfInpMeth == NomCap, calculate the air flow rate specified 
+				// by the NomCap inputs. This overrides all previous values
+				if ( WaterCoil( CoilNum ).CoilPerfInpMeth == NomCap && NomCapUserInp ) {
+					WaterCoil( CoilNum ).InletAirMassFlowRate = WaterCoil( CoilNum ).DesTotWaterCoilLoad / ( CpAirStd*( WaterCoil( CoilNum ).DesOutletAirTemp -
```

Comment:
GetInput tests this for the heating coil and these fields are not autosizable. I do see other instances of a possible divide by 0 for the water coil but those have always been there.

    if ( WaterCoil( CoilNum ).DesInletAirTemp >= WaterCoil( CoilNum ).DesOutletAirTemp ) {
        ShowSevereError( "For " + CurrentModuleObject + ", " + AlphArray( 1 ) );
        ShowContinueError( "  the " + cNumericFields( 5 ) + " must be less than the " + cNumericFields( 7 ) + '.' );
        ErrorsFound = true;
    }